elixirkit/lib/elixirkit/bundler.ex

```diff
@@ -0,0 +1,383 @@
+    cmd("sh", ~w(#{dir}/openssl/build.sh #{cache_dir} #{@openssl_version} macos-x86_64))
+
+    cmd(
+      "sh",
+      ~w(#{dir}/otp/build.sh #{cache_dir} #{@otp_version} macos-universal #{@openssl_version})
+    )
+  end
+
+  @doc false
+  def otp_dir do
+    "#{cache_dir()}/otp-#{@otp_version}-macos-universal"
+  end
+
+  defp build_launcher(release, app_dir, app_name, launcher) do
+    app_tmp = "#{app_dir}/tmp"
+    File.rm_rf!(app_tmp)
+    File.mkdir_p!(app_tmp)
+    File.write!("#{app_tmp}/Launcher.swift", launcher)
+
+    launcher_dir = "#{app_dir}/Contents/MacOS"
+    launcher_path = "#{launcher_dir}/#{app_name}"
+    log(:green, "creating", launcher_path)
+    File.mkdir_p!(launcher_dir)
+    cache_dir = cache_dir()
+
+    targets = [
+      "macos-aarch64",
+      "macos-x86_64"
+    ]
+
+    for target <- targets do
+      otp_dir = "#{cache_dir}/otp-#{@otp_version}-#{target}"
+      cc_target = cc_target(target)
+
+      cmd("clang", [
+        "-c",
+        "-I",
+        "#{otp_dir}/usr/include",
+        "-target",
+        cc_target,
+        "-o",
+        "#{app_tmp}/ElixirKit.o.#{target}",
+        "#{@elixirkit_dir}/c_src/ElixirKit.m"
+      ])
+
+      cmd(
+        "swiftc",
+        [
+          "-parse-as-library",
+          "-import-objc-header",
+          "#{@elixirkit_dir}/c_src/ElixirKit.h",
+          "-lc++",
+          "-ltermcap",
+          "-target",
+          cc_target,
+          "-o",
+          "#{app_tmp}/launcher.#{target}",
+          "#{app_tmp}/Launcher.swift",
+          "#{@elixirkit_dir}/c_src/ElixirKit.swift",
+          "#{otp_dir}/usr/lib/liberl.a",
+          "#{app_tmp}/ElixirKit.o.#{target}"
+        ]
+      )
+
+      # this is a native runner, as opposed to simply a bash script, because the latter seems to
+      # be always executed using Rosetta in an universal app.
+      File.write!(
+        "#{app_tmp}/runner.swift",
+        EEx.eval_file("#{@elixirkit_dir}/c_src/runner.swift", release: release)
+      )
+
+      cmd(
+        "swiftc",
+        ~w(-o #{app_tmp}/runner.#{target} #{app_tmp}/runner.swift -target #{cc_target})
+      )
+    end
+
+    cmd(
+      "lipo",
+      Enum.map(targets, &"#{app_tmp}/launcher.#{&1}") ++ ~w(-create -output #{launcher_path})
+    )
+
+    cmd(
+      "lipo",
+      Enum.map(targets, &"#{app_tmp}/runner.#{&1}") ++ ~w(-create -output #{launcher_path}Runner)
+    )
+
+    erl_path = "#{app_dir}/Contents/Resources/rel/erts-#{release.erts_version}/bin/erl"
+
+    File.write!(erl_path, """
+    #!/bin/sh
+    SELF=$(readlink "$0" || true)
+    if [ -z "$SELF" ]; then SELF="$0"; fi
+    BINDIR="$(cd "$(dirname "$SELF")" && pwd -P)"
+    ROOTDIR="${ERL_ROOTDIR:-"$(dirname "$(dirname "$BINDIR")")"}"
+    EMU=beam
+    PROGNAME=$(echo "$0" | sed 's/.*\\///')
+    export EMU
+    export ROOTDIR
+    export BINDIR
+    export PROGNAME
+    RELEASE_ERLEXEC="${RELEASE_ERLEXEC:-"$BINDIR/erlexec"}"
+    exec $RELEASE_ERLEXEC ${1+"$@"}
```

Comment:
fyi: this is the change from stock `bin/erl`:

```diff
- exec "$BINDIR/erlexec" ${1+"$@"}
+ RELEASE_ERLEXEC="${RELEASE_ERLEXEC:-"$BINDIR/erlexec"}"
+ exec $RELEASE_ERLEXEC ${1+"$@"}
```

Reply:
So, if I get this right, we are still bundling our own ERTS and then we are customizing its `erl` launcher after release? This should be fine because we will have two versions of ERTS anyway, right?

Reply:
Yes, we have two erts: `$rel/erts-$vsn` and `$rel/vendor/otp/erts-$vsn`, the former is a copy of the latter with the only change being exactly the customised `$rel/erts-$vsn/bin/erl` launcher.

Come to think of it, exactly because we have this custom bin/erl launcher, a little bit of duplication maybe is fine. That is, the Livebook itself will use `$rel/erts-$vsn` but Livebook runtimes will use `$rel/vendor/otp/erts-$vsn`.